data upload from s3

kdyz se hitne max retry tak to nevypise nejakou srozumitelnou hlasku

logging
unit tests
custom exceptions